LayoutTests/fast/canvas/drawImage-source-rect-outside-image.html

```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script>
+function runTest() {
+    var canvas = document.getElementById("canvas");
+    var ctx = canvas.getContext("2d");
+    var width = canvas.width;
+    var height = canvas.height;
+
+    var srcCanvas = document.createElement("canvas");
+    srcCanvas.width = width;
+    srcCanvas.height = height;
+    var srcCtx = srcCanvas.getContext("2d");
+
+    srcCtx.arc(width / 2, height / 2, Math.min(width / 2, height / 2), 0, Math.PI * 2)
+    srcCtx.fill();
+    ctx.drawImage(srcCanvas, width / 2, height / 2, width, height, 0, 0, width, height);
```

Comment:
I’m really surprised there is no coverage for this in WPT. Should we put a test in there?

Reply:
Yes, I think so. I'll work with Tim to get a "real" WPT case together.